report/marshal.go

```diff
@@ -0,0 +1,53 @@
+package report
+
+import (
+	"compress/gzip"
+	"encoding/gob"
+	"io"
+
+	"github.com/ugorji/go/codec"
+)
+
+// WriteBinary writes a Report as a gzipped msgpack.
+func (rep Report) WriteBinary(w io.Writer) error {
+	gzwriter, err := gzip.NewWriterLevel(w, gzip.BestCompression)
+	if err != nil {
+		return err
+	}
+	if err = codec.NewEncoder(gzwriter, &codec.MsgpackHandle{}).Encode(&rep); err != nil {
+		return err
+	}
+	gzwriter.Close() // otherwise the content won't get flushed to the output stream
+	return nil
+}
+
+// ReadBinary reads into a Report from a gzipped msgpack.
+//
+// Will decompress the binary if gzipped is true, and will use the given
+// codecHandle to decode it. If codecHandle is nil, will decode as a gob.
```

Comment:
Dropped.

On Tue, 21 Jun 2016 at 10:40 Alfonso Acosta <notifications@github.com>
wrote:

> In report/marshal.go
> <https://github.com/weaveworks/scope/pull/1600#discussion_r67837838>:
>
> > +func (rep Report) WriteBinary(w io.Writer) error {
> > +	gzwriter, err := gzip.NewWriterLevel(w, gzip.BestCompression)
> > +	if err != nil {
> > +		return err
> > +	}
> > +	if err = codec.NewEncoder(gzwriter, &codec.MsgpackHandle{}).Encode(&rep); err != nil {
> > +		return err
> > +	}
> > +	gzwriter.Close() // otherwise the content won't get flushed to the output stream
> > +	return nil
> > +}
> > +
> > +// ReadBinary reads into a Report from a gzipped msgpack.
> > +//
> > +// Will decompress the binary if gzipped is true, and will use the given
> > +// codecHandle to decode it. If codecHandle is nil, will decode as a gob.
>
> I think we can get rid of gob altogether. It's been a long time since
> probes provide msgpack.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/weaveworks/scope/pull/1600/files/81b05a33eea8dfe228d7c205ab3e6dad838c56ef#r67837838>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAHq6vTUUiuSy4M5hPaMf-FMBcvzCGpRks5qN7GAgaJpZM4I4aYm>
> .
>